typescript/ai-sdk-v5/src/plugin-file-parser/file-parser-pdf-url.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * Example: OpenRouter FileParserPlugin - PDF URL (AI SDK)
+ *
+ * This example demonstrates sending PDFs via publicly accessible URLs using AI SDK.
+ * This is more efficient than base64 encoding as you don't need to download
+ * and encode the file.
+ *
+ * Key Points:
+ * - Send PDFs directly via URL without downloading
+ * - Works with AI SDK's file attachment format
+ * - Reduces payload size compared to base64
+ * - Ideal for publicly accessible documents
+ *
+ * To run: bun run typescript/ai-sdk-v5/src/plugin-file-parser/file-parser-pdf-url.ts
+ */
+
+import { createOpenRouter } from '@openrouter/ai-sdk-provider';
+import { generateText } from 'ai';
+
+const openrouter = createOpenRouter({
+  apiKey: process.env.OPENROUTER_API_KEY,
+});
+
+const MODEL = 'anthropic/claude-3.5-sonnet';
+
+/**
+ * Example using the Bitcoin whitepaper (publicly accessible PDF)
+ */
+async function main() {
+  console.log('╔════════════════════════════════════════════════════════════════════════════╗');
+  console.log('║        OpenRouter FileParserPlugin - PDF URL Example (AI SDK)             ║');
+  console.log('╚════════════════════════════════════════════════════════════════════════════╝');
+  console.log();
+  console.log('Sending PDF via public URL (Bitcoin whitepaper)');
+  console.log('URL: https://bitcoin.org/bitcoin.pdf');
+  console.log();
+
+  try {
```

Comment:
Why do we need try/catch here btw if the entry crash it will simply crash right?